crates/viewer/re_time_panel/src/data_density_graph.rs

```diff
@@ -410,6 +410,31 @@ fn smooth(density: &[f32]) -> Vec<f32> {
         .collect()
 }
 
+/// Uniformly sample events using the given sample size.
+///
+/// Each sampled event's count is reweighted to preserve the total density.
+fn uniform_sample_events(events: &[(TimeInt, u64)], sample_size: usize) -> Vec<(TimeInt, u64)> {
+    re_tracing::profile_function!();
+
+    // Use double precision to prevent rounding errors when handling many events.
+    // This ensures that we sample evenly across the entire range.
+    let step = events.len() as f64 / sample_size as f64;
```

Comment:
f32 has 24 bits of precision, so as long as sample_size is < 2^24 we will get even enough sampling with f32 🤷